scalameta/parsers/shared/src/main/scala/scala/meta/internal/parsers/ScalametaParser.scala

```diff
@@ -411,7 +413,8 @@ class ScalametaParser(input: Input)(implicit dialect: Dialect) { parser =>
             while (!sepRegionsProcess.isEmpty &&
               (!sepRegionsProcess.head.isInstanceOf[RegionBrace] &&
                 !sepRegionsProcess.head.isInstanceOf[RegionEnum])) {
-              if (dialect.allowSignificantIndentation) insertOutdent()
+              if (dialect.allowSignificantIndentation && sepRegionsProcess.head.isIndented)
```

Comment:
Added a condition to if, since it caused outdent to appear where none is needed - the parser should have failed at that point also, but the error was misleading.